### PR TITLE
WIP: add 2 matches, update spikes to be float numbers, grab location of th…

### PR DESCRIPTION
**Pull request by MysterionRise**
…e items

### CODE REVIEW

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 1]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
...
-        private val WINDOW_SIZE = 3
-        private val STEP = 1
-        private val DIFFERENCE = 0.2
+        private const val WINDOW_SIZE = 2
+        private const val STEP = 1
+        private const val SMALL_DIFF = 0.1
+        private const val DIFF = 0.2
+        private const val BIG_DIFF = 0.30
+        private val ITEMS_TO_LOOK_FOR = listOf(
+            1, 48, 79, 90, 96, 98, 102, 250, 116, 119, 127, 147, 151, 156, 172, 210, 220, 225, 231, 242, 249, 263
```
Comment (F1nZeR) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: At least comments should be used there to indicate which items are taken into account.

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 2]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
...
+        return abs(v1 - v2) / v2
     }
 
+    private fun analyzeMoving(
+        data: List<Number>,
+        moving: List<Number>
+    ): Float {
+        val zippedData = data.zip(listOf(data[0]) + moving)
+        val isSpike = zippedData.map { it -> relativeDifference(it.first.toDouble(), it.second.toDouble()) > DIFF }
+        println()
```
Comment (F1nZeR) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: empty println

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 3]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
...
+
+    private fun analyzeItems(sortedItems: List<InventoryDistribution>): Float {
+        // TODO produce metrics, saying how spiky items location are
+        return 0.0f
+    }
+
+
     override fun accountFeedback(matches: List<MatchData>, accountId: Long): AccountFeedback {
+        // get items in inventory
+        val item0 = matches.map { it.players.first { it -> it.accountId == accountId }.item0 }
```
Comment (F1nZeR) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: `it.players.first { it -> it.accountId == accountId }` can be moved to a helper method for example.

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 4]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
 
 data class AccountFeedback(
```
Comment (F1nZeR) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: It should contains result decision: is this account is boosted or not. I think probability (percentage) should be used there.

**Review by f1nzer — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 5]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
...
     val accountId: Long,
-    val xpmSpike: Boolean = false,
-    val gpmSpike: Boolean = false,
-    val kpmSpike: Boolean = false,
-    val heroDamageSpike: Boolean = false,
-    val kdaSpike: Boolean = false
+    val xpmSpike: Float = 0.0f,
```
Comment (IgorPerikov) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: I think you can omit the `Float` definition here

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 6]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
...
+        val item4 = matches.map { it.players.first { it -> it.accountId == accountId }.item4 }
+        val item5 = matches.map { it.players.first { it -> it.accountId == accountId }.item5 }
+
+        val distributionPerItem = ITEMS_TO_LOOK_FOR.map { item ->
+            val total = item0.count { it -> it == item } +
+                    item1.count { it -> it == item } +
+                    item2.count { it -> it == item } +
+                    item3.count { it -> it == item } +
+                    item4.count { it -> it == item } +
+                    item5.count { it -> it == item }
```
Comment (IgorPerikov) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: This predicate can be extracted as well

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt [hunk 7]
```diff
@@ -9,48 +9,106 @@ interface AnalysisService {
...
-        val movingAverageGPM =
-            gpmData.windowed(WINDOW_SIZE, STEP) { it.average() }.windowed(WINDOW_SIZE, STEP) { it.average() }
-        val movingAverageKPM =
-            kpmData.windowed(WINDOW_SIZE, STEP) { it.average() }.windowed(WINDOW_SIZE, STEP) { it.average() }
-        val movingAverageHD =
-            heroDamage.windowed(WINDOW_SIZE, STEP) { it.average() }.windowed(WINDOW_SIZE, STEP) { it.average() }
-        val movingAverageKDA =
-            kdaData.windowed(WINDOW_SIZE, STEP) { it.average() }.windowed(WINDOW_SIZE, STEP) { it.average() }
+
+        val movingAverageXPM = xpmData.windowed(WINDOW_SIZE, STEP) { it.average() }
```
Comment (IgorPerikov) [inline src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: this one can be extracted as well, extension function/extracted lambda - you have plenty of choices, as for me I am pretty sure that extension function is a great choice here

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt (hunk 2 above)
Comment (MysterionRise) [reply src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: fu

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt (hunk 5 above)
Comment (MysterionRise) [reply src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: you're not correct

File: src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt (hunk 4 above)
Comment (MysterionRise) [reply src/main/kotlin/com/github/galleyltd/boost/domain/service/AnalysisService.kt]: Agreed, will do however later, after some evaluation of current stuff

**Review by IgorPerikov — APPROVED** [review]

**Review by f1nzer — APPROVED** [review]

**Pull request merged**